### PR TITLE
Added documentation to our site config

### PR DESCRIPTION
**Pull request by GoliathLabs**
I have added some documentation to our site configuration as is the case with the Upstream sample files. I also reformatted both files to have a consistent spacing as suggested by the Upstream files

### CODE REVIEW

**Review by awlx — APPROVED** [review]

File: site.mk [hunk 1]
```diff
@@ -1,337 +1,371 @@
...
 
 ifeq ($(GLUON_TARGET),x86-generic)
-    GLUON_SITE_PACKAGES += $(INCLUDE_PCI) $(INCLUDE_PCI_NET) $(INCLUDE_TLS) $(INCLUDE_USB) $(INCLUDE_USB_NET) $(INCLUDE_USB_SERIAL) $(INCLUDE_USB_STORAGE)
+	GLUON_SITE_PACKAGES += $(INCLUDE_PCI) $(INCLUDE_PCI_NET) $(INCLUDE_TLS) $(INCLUDE_USB) $(INCLUDE_USB_NET) $(INCLUDE_USB_SERIAL) $(INCLUDE_USB_STORAGE)
 
 endif
 
 ifeq ($(GLUON_TARGET),x86-geode)
-    GLUON_SITE_PACKAGES += $(INCLUDE_PCI) $(INCLUDE_PCI_NET) $(INCLUDE_TLS) $(INCLUDE_USB) $(INCLUDE_USB_NET) $(INCLUDE_USB_SERIAL) $(INCLUDE_USB_STORAGE)
+	GLUON_SITE_PACKAGES += $(INCLUDE_PCI) $(INCLUDE_PCI_NET) $(INCLUDE_TLS) $(INCLUDE_USB) $(INCLUDE_USB_NET) $(INCLUDE_USB_SERIAL) $(INCLUDE_USB_STORAGE)
```
Comment (herbetom) [inline site.mk:368]: You noticed that those whitespaces are a result of `contrib/genpkglist.py` and therefore if that script where to run again you would need to fix that again?

Comment (awlx) [reply site.mk:368]: You are right so probably ... this shouldn't be touched

Comment (herbetom) [reply site.mk:368]: Or try to fix it in the source, but just changing the target doesn't seem like the right way to me.

Comment (GoliathLabs) [reply site.mk:368]: Updated the script as well

**Comment by awlx** [comment]
Did you verify that all scripts still work?

**Comment by herbetom** [comment]
And that the output of the script is what you suggested as change?

**Comment by GoliathLabs** [comment]
Yes, the script still works. The spacing is the right length. Of course, since it prints to the console and not directly to a file, they are not tabs, but spaces.

**Comment by goligo** [comment]
Why remove the branch-information from the file? I understand it is not used, as the commit number is also provided, but I still consider it as valuable for someone who wants to update to a later commit, which branch of the repository we are using.

**Comment by GoliathLabs** [comment]
> 
> 
> Why remove the branch-information from the file? I understand it is not used, as the commit number is also provided, but I still consider it as valuable for someone who wants to update to a later commit, which branch of the repository we are using.

You mean the branch information in the module file? I added some explanations in the file itself. This moved the line with the branch to the bottom

**Comment by goligo** [comment]
"# Do not build images for deprecated devices"

This is wrong... should be "Do not build factory images for deprecated devices"

**Comment by GoliathLabs** [comment]
> 
> 
> "# Do not build images for deprecated devices"
> 
> This is wrong... should be "Do not build factory images for deprecated devices"

Fixed

**Comment by goligo** [comment]
> > Why remove the branch-information from the file? I understand it is not used, as the commit number is also provided, but I still consider it as valuable for someone who wants to update to a later commit, which branch of the repository we are using.
> 
> You mean the branch information in the module file? I added some explanations in the file itself. This moved the line with the branch to the bottom

Ah, I see, just reordered, not removed.

**Review by goligo — APPROVED** [review]
I would not change the script, but leave the generated part as is. Next time we update to the latest version of the script, it will get lost again.

**Comment by GoliathLabs** [comment]
> 
> 
> I would not change the script, but leave the generated part as is. Next time we update to the latest version of the script, it will get lost again.

When we implement an updated version of the script, we can simply reformat the file to fit our site configuration.

**Comment by goligo** [comment]
> > I would not change the script, but leave the generated part as is. Next time we update to the latest version of the script, it will get lost again.
> 
> When we implement an updated version of the script, we can simply reformat the file to fit our site configuration.

There is no benefit in changing the indentation of the generated part, but it will cause additional effort in the future.

**Comment by GoliathLabs** [comment]
> 
> 
> > > I would not change the script, but leave the generated part as is. Next time we update to the latest version of the script, it will get lost again.
> > 
> > 
> > When we implement an updated version of the script, we can simply reformat the file to fit our site configuration.
> 
> There is no benefit in changing the indentation of the generated part, but it will cause additional effort in the future.

I don't understand how a simple `gg=G` is effort. We still need to put the generated part in the site.mk file, one way or another. The script is not updated that often, and even if it is updated, we just have to do an automatic reformat, be it with vim, vscode or whatever editor you are on.

**Comment by awlx** [comment]
Changing the script will cause many diffs the next time we add the upstream version. I would refrain from that.

**Pull request merged**